src/array/cuda/rowwise_sampling_prob.cu

```diff
@@ -0,0 +1,667 @@
+/*!
+ *  Copyright (c) 2022 by Contributors
+ * \file array/cuda/rowwise_sampling_prob.cu
+ * \brief weighted rowwise sampling
+ * \author pengqirong (OPPO)
+ */
```

Comment:
Given the amount of content in this file taken from `rowwise_sampling.cu`, we should at least make note of it, if not find someone way to reduce the duplication.

Reply:
Changed accordingly.